### PR TITLE
Issue 13116: should reject returning `this` from a ref class member function

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=13116

### CODE REVIEW

File: changelog.dd [hunk 1]
```diff
@@ -58,6 +59,66 @@ $(LI $(LNAME2 package-protection, $(D package) protection attribute can be optio
     )
 )
 
+$(LI $(LNAME2 rvalue-class-this-super, $(D this) and $(D super) expressions in class method are changed to rvalue)
```
Comment (MartinNowak) [inline changelog.dd]: Should add something like, `$(RED Breaking Change: )` in front of this.

Comment (9rnsr) [inline changelog.dd]: Done.

**Comment by MartinNowak** [comment]
Thanks anyway, returning this remains an error btw, because it escapes a local reference.

**Comment by 9rnsr** [comment]
I still think that passing `this` to a ref parameter should be an error immediately...

**Pull request closed without merging**